### ISSUE TITLE
BIGINT rounded ?

### ISSUE THREAD

**Issue by Adrriii** [mariadb-corporation/mariadb-connector-nodejs#112]
Hello,

I've come accross a very disturbing bug. When performing a simple SELECT on a table with a BIGINT column as the primary key, here is what is printed VS what i have in my database: 

![bug](https://cdn.discordapp.com/attachments/466993499900739604/715584569285410866/unknown.png)

This is a very small set (around 20 entries) and i selected two, which appear in the screenshot. Both are rounded to the hundredth.

The query is as follows : `SELECT * FROM watch_channels WHERE test=1`

The output on the right is logged right after receiving the results from the query, in the callback.

When using `CONVERT(channel USING utf8) as channel`, i get the expected result (in a string, though)

![hack](https://cdn.discordapp.com/attachments/466993499900739604/715588986088980520/unknown.png)

But of course this is a hack, and should be temporary right ?

**Comment by rusher** [comment]
That will be in a future release. 
Javascript use [ double-precision 64-bit binary format IEEE 754](https://en.wikipedia.org/wiki/Floating-point_arithmetic). That mean having exact number until reaching limit of 2^53 - 1. After that, value aren't strictly exact. 

The driver now support 10+ node.js version. 
The driver will at some point support new ES2020 BigInt  but this comes with node.js 10.8.0 only. 

So, that will leave you with this workaround for now ...

**Comment by Adrriii** [comment]
Alright, I hope that helps others if they find the same error, until then !

**Comment by rusher** [comment]
This is now solved with 2.5.0 release with [supportBigInt](https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/master/documentation/connection-options.md#other-options) option. This option can be set on connection or query level. 
This is not enabled by default, however, to avoid major regression. It will be in a future 3.x version. 

This options permit returning  javascript ES2020 [BigInt ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) for [MariaDB/MySQL BIGINT](https://mariadb.com/kb/en/bigint/) data type. 
This ensures having correct value even for value > 2^53 (see [safe range)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER). 

```
await shareConn.query('CREATE TEMPORARY TABLE bigIntTable(id BIGINT)');
await shareConn.query("INSERT INTO bigIntTable value ('9007199254740993')");

const res = await shareConn.query('select * from bigIntTable');
// res :  [{ id: 9007199254740992 }] (wrong value, because not is safe range)

const res2 = await shareConn.query({sql: 'select * from bigIntTable', supportBigInt: true});
// res :  [{ id: 9007199254740993n }] (exact value)
```

https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/master/documentation/promise-api.md#supportbigint
